s3/src/main/scala/akka/stream/alpakka/s3/settings.scala

```diff
@@ -115,7 +116,10 @@ object ForwardProxyCredentials {
 
 }
 
-final class ForwardProxy private (val host: String, val port: Int, val credentials: Option[ForwardProxyCredentials]) {
+final class ForwardProxy private (val host: String,
+                                  val port: Int,
+                                  val credentials: Option[ForwardProxyCredentials],
+                                  val clientTransport: Option[ClientTransport]) {
```

Comment:
This seems like a bit weird way to configure things. In the end, host/port/credentials are one way to specify a `ClientTransport.httpsProxy` and `clientTransport` is a complete alternative. Not sure how this is used exactly and what the compatibility constraints etc are but wouldn't it be better to have a way to make those two ways mutually exclusive?

Reply:
Yes, that would be better. This is mostly a hack to run the tests, but there might be real use cases. I'll have another look.